### PR TITLE
A recipe can be edited

### PR DESCRIPTION
**Pull request by shsteindorsson**
A recipe can be edited
=================
[PT-link](https://www.pivotaltracker.com/story/show/169714712)

## PR content
In this PR we are adding a functionality where a user can update a pre-existing recipe.
```
As a User
In order to improve my own recipe
I would like to be able to change it after creating it
```
## PR content
- unit test
- `update` action for recipe controller

### CODE REVIEW

File: app/controllers/v1/recipes_controller.rb [hunk 1]
```diff
@@ -23,6 +23,15 @@ def show
     render json: recipe, serializer: Recipes::ShowSerializer
   end
 
+  def update
+    recipe = Recipe.find(params[:id])
+    if recipe.update(recipe_params)
+      render json: { message: 'Your recipe has been updated.' }, status: 201
+    else
+      render json: { error_message: 'Unable to update recipe.' }, status: 417
```
Comment (tochman) [inline app/controllers/v1/recipes_controller.rb]: I would go with 422 on this one as well (as we do on create)

Comment (oliverochman) [inline app/controllers/v1/recipes_controller.rb]: recipe.errors.full_messages.to_sentence

File: spec/requests/v1/recipe_can_be_edited_spec.rb [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+      expect(Recipe.last[:title]).to eq 'New Cookies'
+    end
+
+    it "should have new ingredients" do
+      expect(Recipe.last[:title]).to eq 'New Cookies'
+    end
+
+    it "should have new title" do
+      expect(Recipe.last[:title]).to eq 'New Cookies'
+    end
```
Comment (oliverochman) [inline spec/requests/v1/recipe_can_be_edited_spec.rb:34]: please add sad path for too many characters in ingredients

Comment (tochman) [reply spec/requests/v1/recipe_can_be_edited_spec.rb:34]: It is tested in the model

**Review by tochman — APPROVED** [review]
All good

**Pull request merged**